### PR TITLE
Add stream cmd

### PR DESCRIPTION
**Pull request by calvinmvrk**
Add cli cmd "stream" to parse response data

Screenshot 2024-11-07 at 16 33 41

### CODE REVIEW

File: pkg/stream/parse.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	}
+
+	// Check for scanner errors
+	if err := scanner.Err(); err != nil {
+		return fmt.Errorf("error reading file: %w", err)
+	}
+
+	// Print the final concatenated result
+	result := contentBuilder.String()
+	fmt.Println(result)
```
Comment (josebalius) [inline pkg/stream/parse.go:64]: Do you need `result` 😉

File: pkg/stream/parse.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		} else {
+			continue // skip lines without "data: "
+		}
+
+		// Parse the JSON line into our `Data` struct
+		var data Data
+		err := json.Unmarshal([]byte(line), &data)
+		if err != nil {
+			// Skip this line if JSON is incomplete or malformed
+			continue
```
Comment (josebalius) [inline pkg/stream/parse.go:48]: I am actually not sure if we want to silently skip this. I'd want to know if the stream is invalid somehow. You will have to deal with two special case:
1. `data: [DONE]`
2. data line being empty

File: pkg/stream/parse.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		// Check if the line has "data: " prefix
+		if strings.HasPrefix(line, "data: ") {
+			// Remove the "data: " prefix
+			line = strings.TrimPrefix(line, "data: ")
+		} else {
+			continue // skip lines without "data: "
```
Comment (josebalius) [inline pkg/stream/parse.go:40]: For now it's fine, we may want to think in the future of how to handle this for other SSE even types.

**Review by josebalius — APPROVED** [review]
LGTM, left a couple of suggestions

File: pkg/stream/parse.go (hunk 2 above)
Comment (josebalius) [reply pkg/stream/parse.go:56]: Note; I see you made the updates to check for 1 and 2, but we are still silently skipping everything else. If we fail to parse any other data line it should fail and not continue.

Comment (calvinmvrk) [reply pkg/stream/parse.go:56]: Ah, I misunderstood. Will make a new pr to address

**Pull request merged**